addons/l10n_ch/views/account_invoice.xml

```diff
@@ -4,5 +4,12 @@
         <xpath expr="//t[@t-set='show_qr']" position="attributes">
             <attribute name="t-value" add="and o.qr_code_method != 'ch_qr'" separator=" "/>
         </xpath>
+        <xpath expr="//div[hasclass('main_body')]" position="after">
+            <div t-if="not o.l10n_ch_is_qr_valid" class="oe_structure"/>
+            <div t-else="">
+                <div style="page-break-after: always"/>
+                <br/>
+            </div>
+        </xpath>
```

Comment:
I'm a bit afraid about that part:
1) The condition is not the same here and in ir_actions_report.py.
2) What it is not:
```xml
            <div t-if="o.l10n_ch_is_qr_valid">
                <div style="page-break-after: always"/>
                <br/>
            </div>
```

Reply:
1. Indeed, I missed this small part. But after checking, the code was from this [commit](https://github.com/odoo/odoo/commit/736653d2ef63026d0c94762d98e2f282507be39d). I've been able to reproduce the issue it fixes in 16, but not in master, so I don't think this line of code is still relevant. By removing it, the condition would therefore be similar both in the template than in the python file for generating the second page. 

2. This was done like this so that in Studio, the rendering is cleaner, and users can easily add custom stuff above the QR code.